Romaclac msix nonremovableappspolicy (#1761)
MicrosoftDocs/windows-powershell-docs: docset/windows/appx/Get-NonRemovableAppsPolicy.md

---

> author: romaclac

Reviewer: ```suggestion
author: romaclac
```
It seems that your GitHub Username is rdmaclachlan, on author you should use your GH username.

description: Use this topic to help prevent the uninstall of specific Windows apps with Windows PowerShell.
external help file: Microsoft.Windows.Appx.PackageManager.Commands.dll-help.xml
keywords: powershell, cmdlet, msix, appx, block uninstall
manager: jvintzel
Module Name: Appx
ms.assetid: 

> ms.author: romaclac

Reviewer: is this your Microsoft alias right?

ms.date: 02/05/2020
ms.mktglfcycl: manage
ms.prod: w10
ms.reviewer:
ms.sitesec: library
ms.technology: powershell-windows
ms.topic: reference
online version:
schema: 2.0.0
title: Get-NonRemovableAppsPolicy
---

# Get-NonRemovableAppsPolicy

## SYNOPSIS
Gets a list of the app packages that are installed and configured as non-removable apps.

> ## SYNTAX

Reviewer: When there is more than one parameter sets, they are named, for example ByPath or ByOnline, an example https://raw.githubusercontent.com/MicrosoftDocs/office-docs-powershell/master/sharepoint/sharepoint-ps/sharepoint-online/Get-SPOUser.md

Author: I believe I've updated this one correctly.

```powershell
Get-NonRemovableAppsPolicy 
    -Path <string> 
    [-WindowsDirectory <string>] 
    [-SystemDrive <string>] 
    [-LogPath <string>]
    [-ScratchDirectory <string>] 
    [-LogLevel {Errors | Warnings | WarningsInfo}]  
    [<CommonParameters>]
```

```powershell
Get-NonRemovableAppsPolicy 
    -Online 
    [-WindowsDirectory <string>] 
    [-SystemDrive <string>] 
    [-LogPath <string>] 
    [-ScratchDirectory <string>] 
    [-LogLevel {Errors | Warnings | WarningsInfo}]  
    [<CommonParameters>]
```

## DESCRIPTION
The **Get-NonRemovableAppsPolicy** cmdlet gets a list of the app packages what are installed, and have been configured as non-removable (can not be uninstalled). An app package has a .msix or .appx file name extension.

## EXAMPLES

### Example 1: Get all installed non-removable app packages
```
PS C:\> Get-NonRemovableAppsPolicy -Online
```

This command gets information about all installed app packages which have been previously configured as non-removable.

> ### Example 2: Get all non-removable apps from an offline windows image

Reviewer 2: ```suggestion
### Example 2: Get all non-removable apps from an offline Windows image
```

```
PS C:\> Get-NonRemovableAppsPolicy -Path ".\wim\image.wim"
```

This command gets all apps packages that have been loaded into the offline operating system image.

## PARAMETERS

### -Online
Indicates that the cmdlet operates on a running operating system on the local host.

|||
|-|-|
| Type: | SwitchParamter |
| Position: | Named |
| DefaultValue: | None |
| Accept pipeline input: | True (ByPropertyName) |
| Accept wildcard characters: | False |

### -Path

> Specifies the full path to the root directory of the offline Windows image that you will inquire into. If the directory named Windows is not a subdirectory of the root directory, WindowsDirectory must be specified.

Reviewer 3: Remove the leading space.

Author: I'm not sure which space you are referring to?

|||
|-|-|
| Type: | String |
| Position: | Named |
| DefaultValue: | None |
| Accept pipeline input: | True (ByPropertyName) |
| Accept wildcard characters: | False |

### -ScratchDirectory

> Specifies a temporary directory that will be used when extracting files for use during inquiry. The directory must exist locally. If not spcified, the ``` \Windows\%Temp% ``` directory will be used, with a subdirectoryname of a randomly generated hexadecimal value for each run of DISM. Items in the scratch directory are deleted after each operation. You should not use a network share location as a scratch directory to expand a pachage(.cab or .msu file) for installation. The directory used for extracting files for temporary usage during servicing should be a local directory.

Reviewer 3: Fix spelling on 'specified' and change \Windows\Temp to simply `\Windows\Temp` (as that is a static path; use single backticks for in-line code). 'subdirectoryname' should be 'subdirectory name'. 'pachage(' should be 'package ('. The last sentence can be struck as you've stated that network shares cannot be used (which leads the reader to a local directory).

Author: Updated.

> Specifies a temporary directory that will be used when extracting files for use during inquiry. The directory must exist locally. If not spcified, the ``` \Windows\%Temp% ``` directory will be used, with a subdirectoryname of a randomly generated hexadecimal value for each run of DISM. Items in the scratch directory are deleted after each operation. You should not use a network share location as a scratch directory to expand a pachage(.cab or .msu file) for installation. The directory used for extracting files for temporary usage during servicing should be a local directory.

Reviewer 2: ```suggestion
Specifies a temporary directory that will be used when extracting files for use during inquiry. The directory must exist locally. If not spcified, the `\Windows\Temp` directory will be used, with a subdirectory name of a randomly generated hexadecimal value for each run of DISM. Items in the scratch directory are deleted after each operation. You should not use a network share location as a scratch directory to expand a package (.cab or .msu file) for installation.
```

Author: Already updated. But thank you.

|||
|-|-|
| Type: | String |
| Position: | Named |
| DefaultValue: | None |
| Accept pipeline input: | True (ByPropertyName) |
| Accept wildcard characters: | False |

### -SystemDrive
Specifies the path to the location of the BootMgr files. This is necessary only when the BootMgr files are located on a partition other than the one that you are running the command from. Use -SystemDrive to inquire an installed Windows image from a Windows PE environment.

|||
|-|-|
| Type: | String |
| Position: | Named |
| DefaultValue: | None |
| Accept pipeline input: | True (ByPropertyName) |
| Accept wildcard characters: | False |

### -WindowsDirectory
Specifies the path to the Windows directory relative to the image path. This cannot be the full path to the Windows directory; it should be a relative path. If not specified, the default is the Windows directory in the root of the offline image directory.

|||
|-|-|
| Type: | String |
| Position: | Named |
| DefaultValue: | None |
| Accept pipeline input: | True (ByPropertyName) |
| Accept wildcard characters: | False |

### -LogPath

> Specifies the full path and file name to log to. If not set, the default is ``` %WINDIR%\Logs\Dism\dism.log ```. In Windows PE, the default directory is the RAMDISK scratch space which can be as low as 32MB. The log file will automatically be archived. The archived log file will be saved with .bak appended to the file name and a new log file will be generated. Each time the log file is archived the .bak file will be overwritten. When using a network share that is not joined to a domain, use the net use command together with domain credentials to set access permissions before you set the log path for the DISM log.

Reviewer 3: Use single backticks for in-line code. Instead of 'low as', use 'small as'.

Author: Updated.

> Specifies the full path and file name to log to. If not set, the default is ``` %WINDIR%\Logs\Dism\dism.log ```. In Windows PE, the default directory is the RAMDISK scratch space which can be as low as 32MB. The log file will automatically be archived. The archived log file will be saved with .bak appended to the file name and a new log file will be generated. Each time the log file is archived the .bak file will be overwritten. When using a network share that is not joined to a domain, use the net use command together with domain credentials to set access permissions before you set the log path for the DISM log.

Reviewer 2: ```suggestion
Specifies the full path and file name to log to. If not set, the default is `%WINDIR%\Logs\Dism\dism.log`. In Windows PE, the default directory is the RAMDISK scratch space which can be as small as 32MB. The log file will automatically be archived. The archived log file will be saved with .bak appended to the file name and a new log file will be generated. Each time the log file is archived the .bak file will be overwritten. When using a network share that is not joined to a domain, use the net use command together with domain credentials to set access permissions before you set the log path for the DISM log.
```

Author: Had updated previously, but thank you!
